core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/CanonicalizeExpressionRewriter.java

```diff
@@ -251,6 +257,26 @@ public Expression rewriteFunctionCall(FunctionCall node, Void context, Expressio
                 }
             }
 
+            if (functionName.equals("date_trunc")
```

Comment:
it cannot change, since it's user-facing (and per spec, i guess)

eg
```
core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java:    @ScalarFunction("date_trunc")
core/trino-main/src/main/java/io/trino/operator/scalar/time/TimeFunctions.java:    @ScalarFunction("date_trunc")
core/trino-main/src/main/java/io/trino/operator/scalar/timestamp/DateTrunc.java:@ScalarFunction("date_trunc")
core/trino-main/src/main/java/io/trino/operator/scalar/timestamptz/DateTrunc.java:@ScalarFunction("date_trunc")
core/trino-main/src/main/java/io/trino/operator/scalar/timetz/DateTrunc.java:@ScalarFunction("date_trunc")
```